### PR TITLE
filters/modify on join table of m2m relations

### PR DESCRIPTION
**Pull request by falkenhawk**
addressing https://github.com/Vincit/objection.js/issues/1356

Allow specifying `modify` / `filters` for the join table in ManyToManyRelations.

### CODE REVIEW

**Comment by falkenhawk** [comment]
@koskimas would you consider this for objection v3.0? I could try to rebase and add tests, but not sure if I could make it on time, depends when you would like to move out of v3 alpha stage

**Comment by falkenhawk** [comment]
@koskimas I've added some tests and adjusted docs.

File: lib/queryBuilder/QueryBuilder.js [hunk 1]
```diff
@@ -325,7 +325,7 @@ class QueryBuilder extends QueryBuilderBase {
         );
       } else if (op.constructor === FromOperation) {
         return op.table === tableName;
-      } else if (op.name === 'as' || op.is(FindOperation)) {
+      } else if (op.name === 'as' || op.is(FindOperation) || op.is(OnErrorOperation)) {
```
Comment (falkenhawk) [inline lib/queryBuilder/QueryBuilder.js:328]: I have no recollection though why I added this.

Comment (lehni) [reply lib/queryBuilder/QueryBuilder.js:328]: It's not wrong: If you have a query that only contains an `onError()` operation, it's still a select-all query.
As for why it was added, I'd guess you had some logic in your code base that always calls `onError()` o your queries and wanted to filter this out? So I do think this makes sense, but it's not really in the scope of this PR to add it?

**Comment by lehni** [comment]
LGTM, thanks @falkenhawk!

As for the `OnErrorOperation` bit, I think it's not wrong to leave this in. Merging.

**Comment by developer239** [comment]
The new API is not doing anything and both `filter` and `modify` are ignored.

**Pull request merged**